File: Data Structures & Algorithms/meeting-schedule-ii/submission-2.py
"""
Definition of Interval:
class Interval(object):
    def __init__(self, start, end):
        self.start = start
        self.end = end
"""

class Solution:
    def minMeetingRooms(self, intervals: List[Interval]) -> int:
        start = sorted([i.start for i in intervals])
        end = sorted([i.end for i in intervals])
        res = count = 0

        startI = 0
        endI = 0
        
        while startI < len(intervals):
            if start[startI] < end[endI]:
                count += 1
                res = max(count, res)
                startI += 1
            else:
                count -= 1
                endI += 1
        
        return res

File: Data Structures & Algorithms/counting-bits/submission-2.java
class Solution {
    public int[] countBits(int n) {
        int[] dp = new int[n + 1];
        for(int i = 1; i <= n; ++i) {
            dp[i] = dp[i >> 1] + (i & 1);
        }
        return dp;
    }
}


// 0   0
// 01  1
// 10  2
// 11  3
// 100 4
// 101 5
// 110 6
// 111 7
